Trakt/Extensions.cs

```diff
@@ -234,26 +248,181 @@ public static IList<IEnumerable<T>> ToChunks<T>(this IEnumerable<T> enumerable,
+            {
+                return true;
+            }
+
+            var imdb = item.GetProviderId(MetadataProvider.Imdb);
+            if (!string.IsNullOrEmpty(imdb) && string.Equals(imdb, show.Ids.Imdb, StringComparison.OrdinalIgnoreCase))
+            {
+                return true;
+            }
+
+            var tvrage = item.GetProviderId(MetadataProvider.TvRage);
+            if (!string.IsNullOrEmpty(tvrage) && string.Equals(tvrage, show.Ids.Tvrage, StringComparison.OrdinalIgnoreCase))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Checks if a <see cref="Episode"/> matches a <see cref="TraktEpisode"/>.
+        /// </summary>
+        /// <param name="item">The <see cref="Episode"/>.</param>
+        /// <param name="episode">The <see cref="TraktEpisode"/>.</param>
+        /// <returns><see cref="bool"/> indicating if the <see cref="Episode"/> matches a <see cref="TraktEpisode"/>.</returns>
+        public static bool IsMatch(Episode item, TraktEpisode episode)
+        {
+            var tvdb = item.GetProviderId(MetadataProvider.Tvdb);
+            if (!string.IsNullOrEmpty(tvdb) && string.Equals(tvdb, episode.Ids.Tvdb, StringComparison.OrdinalIgnoreCase))
+            {
+                return true;
+            }
+
+            var tmdb = item.GetProviderId(MetadataProvider.Tmdb);
+            if (!string.IsNullOrEmpty(tmdb) && string.Equals(tmdb, episode.Ids.Tmdb.ToString(), StringComparison.OrdinalIgnoreCase))
+            {
+                return true;
+            }
+
+            var imdb = item.GetProviderId(MetadataProvider.Imdb);
+            if (!string.IsNullOrEmpty(imdb) && string.Equals(imdb, episode.Ids.Imdb, StringComparison.OrdinalIgnoreCase))
+            {
+                return true;
+            }
+
+            var tvrage = item.GetProviderId(MetadataProvider.TvRage);
+            if (!string.IsNullOrEmpty(tvrage) && string.Equals(tvrage, episode.Ids.Tvrage, StringComparison.OrdinalIgnoreCase))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Checks if a <see cref="Episode"/> matches a <see cref="TraktEpisodeWatched"/>.
+        /// </summary>
+        /// <param name="item">The <see cref="Episode"/>.</param>
+        /// <param name="episode">The <see cref="TraktEpisodeWatched"/>.</param>
+        /// <returns><see cref="bool"/> indicating if the <see cref="Episode"/> matches a <see cref="TraktEpisodeWatched"/>.</returns>
+        public static bool IsMatch(Episode item, TraktEpisodeWatched episode)
+        {
+            var episodeNumber = episode.Number;
+            var itemIndexNumber = item.IndexNumber;
+            var itemIndexNumberEnd = item.IndexNumberEnd;
+
+            return !itemIndexNumber.HasValue
```

Comment:
`Episode.ContainsEpisodeNumber(int number)` already does this

Reply:
Nice, didn't know that. Seems like we can replace custom code in some other places too.